sphinxcontrib/drawio/__init__.py

```diff
@@ -158,14 +158,17 @@ def drawio_export(
     sha_key = sha1(hash_key.encode()).hexdigest()
     filename = Path(input_stem).with_suffix("." + output_format)
     export_relpath = Path(".drawio") / sha_key / filename
-    export_abspath = Path(builder.srcdir) / export_relpath
+    if builder.config.drawio_builder_export_directory == "outdir":
+        export_abspath = Path(builder.outdir) / export_relpath
```

Comment:
I feel _builder.outdir_ is not the correct location for this, since the `.drawio` directory can be shared by all builders, similar to the doctrees directory.

When building using the Makefile, _builder.outdir_ points to `_build/html` for example. The doctrees go into `_build/doctrees`. A good drawio-export location would therefore be `_build/drawio`. However, when not using the Makefile, you need to specify the output directory as an argument to _sphinx-build_ and there is no guarantee that this directory hierarchy exists.

If the [-d option](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-d) is not passed to sphinx-build, the doctrees are stored in _a directory called .doctrees under the build directory_. So perhaps the way to go is to do the same for the drawio-export directory:
- if _drawio_export_directory_ is not set, use _builder.outdir_`/.drawio` (like you did)
- if _drawio_export_directory_ is set, it should specify a path that will be shared by all builders

Too bad that won't automatically use `_build/drawio` when using the Makefile. Perhaps we can try to detect this (and equivalent) situation(s) by comparing _builder.outdir_ to _builder.doctreedir_ to detemine the location for the drawio-export directory. Or is that too much magic? Comments welcome!

Reply:
Thanks for the comment.

I changed the option as you suggested.
It defaults to builder.outdir (this is the same behaviour as the plantuml integration)
Using the new option you can specify one directory for all exports to reuse them across multiple builders.

I would keep it simple without too much magic.
In most cases I guess you would build pdf and html which usually use different formats anyway.
Using the option you would be able to tweak the build time(and disk usage) in these special cases by specifying a central output directory.